src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -11745,6 +11749,125 @@ def str_len(self, **kwargs: Any) -> "SnowflakeQueryCompiler":
             self._modin_frame.apply_snowpark_function_to_data_columns(length)
         )
 
+    def str_slice(
+        self,
+        start: Optional[int] = None,
+        stop: Optional[int] = None,
+        step: Optional[int] = None,
+    ) -> "SnowflakeQueryCompiler":
+        """
+        Slice substrings from each element in the Series or Index.
+
+        Parameters
+        ----------
+        start : int, optional
+            Start position for slice operation.
+        stop : int, optional
+            Stop position for slice operation.
+        step : int, optional
+            Step size for slice operation.
+
+        Returns
+        -------
+        SnowflakeQueryCompiler representing result of the string operation.
+        """
+        if step == 0:
+            raise ValueError("slice step cannot be zero")
```

Comment:
I don't think you need this check here since this is done in `slice()` already.

Reply:
Good point. Removed!